controllers/comment.js

```diff
@@ -1,23 +1,60 @@
 import Sequelize from 'sequelize';
+import { format } from 'date-fns';
 import models from '../models';
 
 const { iLike } = Sequelize.Op;
 const { Article, Comment } = models;
 
 export default {
   async addComment(req, res) {
-    const article = await Article.findOne({
-      where: { slug: { [iLike]: req.params.slug } }
-    });
+    const currentTime = format(new Date(), 'MMMM Do YYYY, h:mm:ss a');
+    const { params: { slug }, decoded } = req;
+    const article = await Article
+      .findOne({ where: { slug: { [iLike]: slug } } });
     const newComment = await Comment.create({
-      userId: req.decoded.id,
-      comment: req.body.comment,
+      userId: decoded.id,
+      comment: { [currentTime]: req.body.comment },
       articleId: article.id
     });
-    return res.json({
-      comment: newComment.comment,
-      id: newComment.id,
-      updatedAt: newComment.updatedAt
+    const comment = Object.values(newComment.comment)[0];
+    const { id, updatedAt } = newComment;
+    return res.json({ comment, id, updatedAt });
+  },
+
+  async editComment(req, res) {
+    const currentTime = format(new Date(), 'MMMM Do YYYY, h:mm:ss a');
+    const { params: { id }, decoded, body: { newComment } } = req;
+    const commentRow = await Comment
+      .findOne({ where: { id, userId: decoded.id } });
+    const { comment } = commentRow;
+    comment[currentTime] = newComment;
+    await commentRow.update({ comment });
+    const updatedComment = Object.values(comment).slice(-1)[0];
+    return res.json({ updatedComment });
+  },
+
+  async getComment(req, res) {
+    const { id } = req.params;
+    const commentRow = await Comment.findOne({ where: { id } });
+    const { comment } = commentRow;
+    const latestEdit = Object.values(comment).slice(-1)[0];
```

Comment:
What is the `slice`used for here?

Reply:
The `slice(-1)` returns an array of just the last value, and then `[0]` returns the value itself